File: FreStyle/src/main/java/com/example/FreStyle/entity/ScenarioBookmark.java
package com.example.FreStyle.entity;

import java.sql.Timestamp;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import jakarta.persistence.UniqueConstraint;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "scenario_bookmarks", uniqueConstraints = {
    @UniqueConstraint(columnNames = {"user_id", "scenario_id"})
})
@Data
@NoArgsConstructor
@AllArgsConstructor
public class ScenarioBookmark {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @ManyToOne
    @JoinColumn(name = "scenario_id", nullable = false)
    private PracticeScenario scenario;

    @Column(name = "created_at", insertable = false, updatable = false)
    private Timestamp createdAt;
}

File: backend/internal/domain/scenario_bookmark.go
package domain

import "time"

// ScenarioBookmark は練習シナリオへのブックマーク。
type ScenarioBookmark struct {
	ID         uint64    `gorm:"primaryKey" json:"id"`
	UserID     uint64    `gorm:"column:user_id;index" json:"userId"`
	ScenarioID uint64    `gorm:"column:scenario_id;index" json:"scenarioId"`
	CreatedAt  time.Time `gorm:"column:created_at" json:"createdAt"`
}

func (ScenarioBookmark) TableName() string { return "scenario_bookmarks" }
